mcs/build/library.make

```diff
@@ -28,11 +28,23 @@ LIB_MCS_FLAGS += $(patsubst %,-r:%.dll, $(subst =,=$(topdir)/class/lib/$(PROFILE
 
 sourcefile = $(LIBRARY).sources
 
+profile_library = $(PROFILE)_$(LIBRARY)
+
 # If the directory contains the per profile include file, generate list file.
+ifdef intermediate
+intermediate_name = $(patsubst %_,%,$(subst /,_,$(intermediate)))
+PROFILE_sources := $(wildcard $(PROFILE)_$(intermediate_name)_$(LIBRARY).sources)
+ifdef PROFILE_sources
```

Comment:
We should use the new naming scheme everywhere, then we can get rid of of the conditional fallback.